### ISSUE TITLE
Remove meaningless 0-byte writes for every log message

### ISSUE THREAD

**Issue by cheapRoc** [joyent/containerpilot#540]
Ref: #426 (SWSUP-955 and SWSUP-1002)

It was observed by at least two customers of logging issues with recent releases of ContainerPilot. In either case dropped logs or missing logs entirely. Only LX brand zones under SmartOS were effected.

The following Dtrace shows meaningless zero byte writes being output for every log message in ContainerPilot (`'\0'`).

```
93085663714502 7db11f57 containerpilot[43151] lx-syscall::write:entry(1,/dev/zfd/1)
93085663759611 7db11f57 containerpilot[43151] lx-syscall::write:return(1,/dev/zfd/1) = '2018-01-03T20:31:14.03914014Z     2018-01-03 20:31:14 cp-log: -- Begin /tmp/outputs/test.log\n\0'[93]
93085663769623   global zoneadmd[42972] read(14[/devices/pseudo/zfdnex@2/zfd@6:master6],1024):entry
93085663788297 7db11f57 containerpilot[43151] lx-syscall::write:entry(1,/dev/zfd/1)
93085663789728   global zoneadmd[42972] read(14[/devices/pseudo/zfdnex@2/zfd@6:master6]):return = '2018-01-03T20:31:14.03914014Z     2018-01-03 20:31:14 cp-log: -- Begin /tmp/outputs/test.log\n\0'[93]
93085663816641 7db11f57 containerpilot[43151] lx-syscall::write:return(1,/dev/zfd/1) = '\0'[0]
93085663846775   global zoneadmd[42972] read(14[/devices/pseudo/zfdnex@2/zfd@6:master6],1024):entry
93085663862639 global zoneadmd[42972] read(14[/devices/pseudo/zfdnex@2/zfd@6:master6]):return = '\0'[0]
```

The 0-byte writes were introduced by [this commit](https://github.com/joyent/containerpilot/commit/8abeb51cb1b69f201b2dd2af73182a6d3af41187) which was an attempt by a community member to standardize our log timestamps to RFC 3339. The redundant writes were included unbeknown to either the contributor or the reviewer.

Typically these are harmless but they caused an issue in SmartOS specifically with `zoneadm`. This was subsequently patched in [OS-6521](https://github.com/joyent/illumos-joyent/commit/544f34505a03cd85e0b4d03c9b3d3add73c025ed).

We need to remove these 0-byte writes as they are not providing anything meaningful. Reverting the commit fixes the problem but I'd like to maintain the timestamp formatting. I'm exploring a patch now.

**Comment by mterron** [comment]
Are the logs being emitted with a duplicate date? I remember trying to get rid of it by creating a formatter as per the Go documentation.

```
2018-01-03T20:31:14.03914014Z     2018-01-03 20:31:14 cp-log: -- Begin /tmp/outputs/test.log

**Comment by cheapRoc** [comment]
Correct, I realized later that the log output itself was including the date and not duplicating it.

We can actually get away with re-applying Logrus and allowing it to format log output. This would revert the existing unstructured log output so I'm also looking into how I can implement just what we have today without `fmt.Print` outside logrus. 

The following gives us both worlds but I'm unsure if we should switch back to logrus output again. Reading old PRs and issues for relevancy.

```go
formatter = &logrus.TextFormatter{
    DisableColors:   true,
    FullTimestamp:   true,
    TimestampFormat: time.RFC3339Nano,
}
```

**Comment by cheapRoc** [comment]
Using Logrus again gives us the following.

With `DisableColors` set to `false`.

```
INFO[2018-01-05T14:59:02.569894235-05:00] control: serving at /tmp/cp-args.socket
INFO[2018-01-05T14:59:02.576709786-05:00] Missing -text option!                         job=goocar pid=18036
ERRO[2018-01-05T14:59:02.578944836-05:00] goocar exited with error: exit status 127
```

With `DisableColors` set to `true` we always get the non-TTY format.

```
time="2018-01-05T14:49:23.305935644-05:00" level=info msg="control: serving at /tmp/cp-args.socket"
time="2018-01-05T14:49:23.309448857-05:00" level=info msg="Missing -text option!" job=goocar pid=6099
time="2018-01-05T14:49:23.309648422-05:00" level=error msg="goocar exited with error: exit status 127"
```

This is what we have today...

```
2018-01-05T14:49:44.585880702-05:00 control: serving at /tmp/cp-args.socket
2018-01-05T14:49:44.596279198-05:00 Missing -text option!
2018-01-05T14:49:44.599186053-05:00 goocar exited with error: exit status 127
```

I'm unsure as to why we switched to also use the Go standard library and why we're subverting Logrus formatting. [This commit](https://github.com/joyent/containerpilot/commit/6c4f782d6b3d1acca6f8bdc1e07c210ee59c83c4) introduced the `DefaultLogFormatter` which removes all formatting. My question is, why did we go back to unstructured output?

**Comment by cheapRoc** [comment]
Actually, reading a bit more I guess it was because `text` outputs using the `TextFormatter` already. I'll see if I can output the simplified version. The problem is we can't subvert Logrus so we need to implement the `logrus.Formatter` type and also handle the buffer properly.

**Comment by cheapRoc** [comment]
Looks like my PR still holds water than. Here's the output...

```
2018-01-05T15:10:01.095792962-05:00 control: serving at /tmp/cp-args.socket
2018-01-05T15:10:01.103699065-05:00 Missing -text option!
2018-01-05T15:10:01.106360144-05:00 goocar exited with error: exit status 127
```

Performing final testing now.

**Comment by cheapRoc** [comment]
Patched in release [3.6.2](https://github.com/joyent/containerpilot/releases/tag/3.6.2).
